### PR TITLE
Added NUCLEO_F429ZI support

### PR DESCRIPTION
**Pull request by capriele**

### CODE REVIEW

File: stmhal/boards/NUCLEO_F429ZI/mpconfigboard.h [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+#define MICROPY_HW_ENABLE_TIMER     (1)
+#define MICROPY_HW_ENABLE_SERVO     (0)
+#define MICROPY_HW_ENABLE_DAC       (0)
+#define MICROPY_HW_ENABLE_CAN       (1)
+
+// HSE is 8MHz
+#define MICROPY_HW_CLK_PLLM (8)
+#define MICROPY_HW_CLK_PLLN (180)
+#define MICROPY_HW_CLK_PLLP (RCC_PLLP_DIV2)
+#define MICROPY_HW_CLK_PLLQ (4)
```
Comment (dpgeorge) [inline stmhal/boards/NUCLEO_F429ZI/mpconfigboard.h]: Does this configure the CPU clock to 180MHz?  Is that the maximum?

File: stmhal/boards/NUCLEO_F429ZI/mpconfigboard.h [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+// From the reference manual, for 2.7V to 3.6V
+// 151-180 MHz => 5 wait states
+// 181-210 MHz => 6 wait states
+// 211-216 MHz => 7 wait states
+#define MICROPY_HW_FLASH_LATENCY    FLASH_LATENCY_6
```
Comment (dpgeorge) [inline stmhal/boards/NUCLEO_F429ZI/mpconfigboard.h]: Could it use a latency of 5 if the freq is 180MHz?

File: stmhal/boards/NUCLEO_F429ZI/mpconfigboard.h [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+#define MICROPY_HW_UART3_RX     (pin_D9)
+
+#define MICROPY_HW_UART4_TX     (pin_A0)
+#define MICROPY_HW_UART4_RX     (pin_C11)
+
+#define MICROPY_HW_UART5_TX     (pin_C12)
+#define MICROPY_HW_UART5_RX     (pin_D2)
+
+#define MICROPY_HW_UART_REPL        PYB_UART_3
+#define MICROPY_HW_UART_REPL_BAUD   115200
```
Comment (dpgeorge) [inline stmhal/boards/NUCLEO_F429ZI/mpconfigboard.h]: Did you test that the REPL works on this UART?

File: stmhal/boards/NUCLEO_F429ZI/stm32f4xx_hal_conf.h [hunk 4]
```diff
@@ -0,0 +1,412 @@
...
+
+#ifdef __cplusplus
+ extern "C" {
+#endif
+
+/* Exported types ------------------------------------------------------------*/
+/* Exported constants --------------------------------------------------------*/
+
+#define USE_USB_HS
+#define USE_USB_HS_IN_FS
```
Comment (dpgeorge) [inline stmhal/boards/NUCLEO_F429ZI/stm32f4xx_hal_conf.h]: I assume this setting is necssary?

**Review by dpgeorge — COMMENTED** [review]
Thanks for the contribution.  What kind of testing did you do?

File: stmhal/boards/NUCLEO_F429ZI/mpconfigboard.h (hunk 1 above)
Comment (capriele) [inline stmhal/boards/NUCLEO_F429ZI/mpconfigboard.h]: Yes 180Mhz is the maximum clock

File: stmhal/boards/NUCLEO_F429ZI/mpconfigboard.h (hunk 3 above)
Comment (capriele) [inline stmhal/boards/NUCLEO_F429ZI/mpconfigboard.h]: Yes, I've tried

**Comment by dpgeorge** [comment]
Ok, thanks for the confirmation, merged in 5b509dbc7bd542ad9fdc7e94f8b8c096db8bef40

**Pull request closed without merging**